### ISSUE TITLE
Visualizer not visualizing audio with a url

### ISSUE THREAD

**Issue by roseaneesha** [staskobzar/vue-audio-visual#124]
The audio plays, but the visualizer doesn't seem to show the waves. 
This only happens if the source audio is a url (eg:https://www.soundjay.com/human/sounds/heartbeat-01a.mp3) 
Works perfectly if the audio is added as a file in the local system

![Screenshot_20221218_210258](https://user-images.githubusercontent.com/72336895/208306891-f1fb12f8-913e-471b-9c85-d1e9a084e597.png)

**Comment by staskobzar** [comment]
It is blocked by CORS policy. I will work on the patch.
Thanks for the reports

**Comment by staskobzar** [comment]
@roseaneesha I have made some improvements in the module to make sure CORS is enabled during fetch.
But the main problem you have is that you source is not returning "Access-Control-Allow-Origin" header, thus restricting access to the source from ajax requests

**Comment by clemenceroumy** [comment]
👋🏻 Hello @staskobzar, I still have trouble using the component with a spotify url source audio (ex: https://p.scdn.co/mp3-preview/805bbe52bc7f8412e9027579787251375e6b847d?cid=11381bbca5f3479f9462199118c15ad9)
Is there any way to make this work ?

**Comment by staskobzar** [comment]
Hello @clemenceroumy 
CORS access controll can be set with "<audio>" element attribute "crossorigin".
When you use high level interface functions like "```AVLine(...)```" this attribute is not enabled. You can use composable function to build and  access your "<audio>" element.

To make your link work with AVLine you would do something like that:
```vue
<script setup lang="ts">
import { ref } from 'vue'
import { makeLineProps } from '@/composables/useProps'
import { useAVLine } from '@/composables/useAVLine'

const props = defineProps(makeLineProps())
const player = ref(null)
const canvas = ref(null)

const spotifyURL = ref('https://p.scdn.co/mp3-preview/805bbe52bc7f8412e9027579787251375e6b847d?cid=11381bbca5f3479f9462199118c15ad9')

useAVLine(player, canvas, props)
</script>

<template>
  <audio ref="player"
    crossorigin="anonymous"
    :controls="true"
    :src="spotifyURL" />
  <canvas ref="canvas" />
</template>
```

Note crossorigin value in audio element:
![image](https://github.com/staskobzar/vue-audio-visual/assets/147280/186c5f99-327f-453d-9e2a-03d34932681d)

Basically, all high level functions use compable function to [build plugin](https://github.com/staskobzar/vue-audio-visual/tree/master#gear-api). You can check "[src/components](https://github.com/staskobzar/vue-audio-visual/tree/master/src/components)" folder to see how it works for all the components.

**Comment by clemenceroumy** [comment]
@staskobzar Oh yeah, that works well ! Sorry for the inconveniance and thank you for your quick reply and work :)

**Comment by staskobzar** [comment]
@drewjgray v2 has is
![image](https://github.com/staskobzar/vue-audio-visual/assets/147280/2a9ff4ed-42e2-47a1-8513-be99ee6ca78b)

https://github.com/staskobzar/vue-audio-visual/tree/v2#common-props
